### DISCUSSION TITLE
Inserting nuclei in phase field model

### DISCUSSION THREAD

**Discussion by vermaprk** [idaholab/moose#24947] [Q&A Modules: Phase field] [answered]
Hello, I am trying to insert nuclei in my phase-field model using the discrete nucleation system. I am using the free energy penalty based nucleation method. As per my understanding, the `DiscreteNucleation` material property forces the variable to a certain desired value using simple harmonic form penalty, which affects the overall free energy of the system, thus allowing nuclei formation.
therefore, total free energy = actual free energy + nucleation free energy penalty
I have following confusions related to my problem which comprises of two variables eta (phase order parameter) and mu (chemical potential). I am trying to force mu to a desired value using `DiscreteNucleation` which will cause the change in order parameter via change in total free energy.

1. My free energy term (without the nucleation penalty) already contains mu. So when I force mu to a desired value, it already gets modified. Then, why I have to add the `DiscreteNucleation` term again by using `DerivativeSumMaterial` material property.
2. I have tried different values of mu but this doesn't causes any change in eta.
3. I tried forcing the eta term to 1 instead of mu. This works but I am not sure whether this is correct approach. 
4. Another issue is related to hold time. Once the hold time is over the simulation undergoes convergence issues. Is increasing the hold time to simulation time a good approach for this issue ?
5. The nucleation rate which is introduced using userobject `DiscreteNucleationInserter` has probability. So, is this probability the no. of nuclei inserted per unit area per unit time or anything else between 0 to 1.
Lastly please refer some materials to read about the free energy penalty based nucleation.

Thanks

**Comment by GiudGiud** [comment]
@amjokisaari @dschwen on phase nucleation please

**Reply by vermaprk** [reply]
@laagesen on nuclei insertion in PFM

**Comment by dschwen** [comment, marked as answer]
> 1. My free energy term (without the nucleation penalty) already contains mu. So when I force mu to a desired value, it already gets modified. Then, why I have to add the DiscreteNucleation term again by using DerivativeSumMaterial material property.

You need to add the penalty term (mu-mu0)^2

> 2. I have tried different values of mu but this doesn't causes any change in eta.

Is this a grand potential model? You might want to force a change to eta instead. The key is to apply the change forcing a change in a certain field variable and then let the system evolve towards a lower energy state by diffusion of solute.

> 3. I tried forcing the eta term to 1 instead of mu. This works but I am not sure whether this is correct approach.

Yes. See 2.

> 4. Another issue is related to hold time. Once the hold time is over the simulation undergoes convergence issues. Is increasing the hold time to simulation time a good approach for this issue ?

The hold time is related to the diffusional timescale of the system. You can monitor the penalty term by adding `outputs = exodus` to the discrete nucleation material. Pay attention to how long it takes for the penalty field to go to approximately zero. This should be your hold time.

> 5. The nucleation rate which is introduced using userobject DiscreteNucleationInserter has probability. So, is this probability the no. of nuclei inserted per unit area per unit time or anything else between 0 to 1.
Lastly please refer some materials to read about the free energy penalty based nucleation.

It is the probability to insert a nucleus per unit area and unit time. Check out the `DiscreteNucleationTimeStep` post processor for setting a reasonable time step limit, and the `DiscreteNucleationMarker` for pre-refining the mesh before nucleus insertion.

I never got to writing a stand alone paper about the free energy based nucleation system, but it was featured in https://www.sciencedirect.com/science/article/pii/S0927025622004839

**Reply by vermaprk** [reply]
Thanks
